### ISSUE TITLE
get NullPointerException when using BlockHound in springwebflux

### ISSUE THREAD

**Issue by haicoder** [reactor/BlockHound#10]
get NullPointerException when I add `BlockHound.install();` in spring webflux :

```
@SpringBootApplication
@EnableCaching
public class DemoApplication{

    public static void main(String[] args) {
        BlockHound.install();

        System.setProperty("reactor.netty.http.server.accessLogEnabled", "true");
        SpringApplication.run(DemoApplication.class, args);
    }
}
```

Exception: 
```
java.lang.NullPointerException
	at reactor.BlockHoundRuntime.<clinit>(BlockHoundRuntime.java:31)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at reactor.BlockHound$Builder.install(BlockHound.java:227)
	at reactor.BlockHound.install(BlockHound.java:63)
	at com.huaweicloud.resourceprovider.openstack.DemoApplication.main(DemoApplication.java:16)
Exception in thread "main" java.lang.RuntimeException: java.lang.ExceptionInInitializerError
	at reactor.BlockHound$Builder.install(BlockHound.java:276)
	at reactor.BlockHound.install(BlockHound.java:63)
	at com.huaweicloud.resourceprovider.openstack.DemoApplication.main(DemoApplication.java:16)
Caused by: java.lang.ExceptionInInitializerError
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at reactor.BlockHound$Builder.install(BlockHound.java:227)
	... 2 more
Caused by: java.lang.RuntimeException: java.lang.NullPointerException
	at reactor.BlockHoundRuntime.<clinit>(BlockHoundRuntime.java:38)
	... 7 more
Caused by: java.lang.NullPointerException
	at reactor.BlockHoundRuntime.<clinit>(BlockHoundRuntime.java:31)
	... 7 more
```

Can you tell me where it is causing the error?

**Comment by bsideup** [comment]
Hi @haicoder,

Which OS are you using?

Could you please set a breakpoint in `BlockHoundRuntime`'s `static` init block?
I have a feeling that `mapLibraryName` on your OS returns a suffix which is not supported yet by BlockHound (native lib), but we can add it.

**Comment by haicoder** [comment]
Hi @bsideup , thanks for your help

I am using centos7.

I copy some code from `BlockHoundRuntime`'s `static` init block :

```
    public static void main(String[] args) {
        try {
            Path tempFile = Files.createTempFile("BlockHound", ".dylib");
            System.out.println(tempFile.toUri());
            try (InputStream inputStream = BlockHoundRuntime.class.getResource("/" + System.mapLibraryName("BlockHound")).openStream()) {
                System.out.println(BlockHoundRuntime.class.getResource("/" + System.mapLibraryName("BlockHound")));
                Files.copy(inputStream, tempFile, StandardCopyOption.REPLACE_EXISTING);
            }
            System.load(tempFile.toAbsolutePath().toString());
        } catch (Throwable e) {
            e.printStackTrace();
            throw new RuntimeException(e);
        }
        System.out.println("begin");
        BlockHound.install();

        System.setProperty("reactor.netty.http.server.accessLogEnabled", "true");
        SpringApplication.run(OpenstackProviderApplication.class, args);
    }
```

and I can get the path of 'libBlockHound.so', this is output: 
```
[root@localhost test]# /usr/java/jdk1.8.0_201-amd64/bin/java -jar demo.jar
file:///tmp/BlockHound3781531311566606212.dylib
jar:file:/tmp/test/install.jar!/BOOT-INF/lib/blockhound-1.0.0.M1.jar!/libBlockHound.so
begin
21:06:23.404 [main] DEBUG reactor.util.Loggers$LoggerFactory - Using Slf4j logging framework
java.lang.NullPointerException
	at reactor.BlockHoundRuntime.<clinit>(BlockHoundRuntime.java:31)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at reactor.BlockHound$Builder.install(BlockHound.java:227)
	at reactor.BlockHound.install(BlockHound.java:63)
	at com.huaweicloud.resourceprovider.openstack.OpenstackProviderApplication.main(OpenstackProviderApplication.java:37)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
Caused by: java.lang.RuntimeException: java.lang.ExceptionInInitializerError
	at reactor.BlockHound$Builder.install(BlockHound.java:276)
	at reactor.BlockHound.install(BlockHound.java:63)
	at com.huaweicloud.resourceprovider.openstack.OpenstackProviderApplication.main(OpenstackProviderApplication.java:37)
	... 8 more
Caused by: java.lang.ExceptionInInitializerError
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at reactor.BlockHound$Builder.install(BlockHound.java:227)
	... 10 more
Caused by: java.lang.RuntimeException: java.lang.NullPointerException
	at reactor.BlockHoundRuntime.<clinit>(BlockHoundRuntime.java:38)
	... 15 more
Caused by: java.lang.NullPointerException
	at reactor.BlockHoundRuntime.<clinit>(BlockHoundRuntime.java:31)
	... 15 more
```

**Comment by bsideup** [comment]
Hi @haicoder,

Interesting. It seems that it fails to `getResource` when re-packaged into Spring Boot jar.

Thanks for checking, I will debug it right away and push a snapshot with a fix. Sorry for that!

**Comment by haicoder** [comment]
Thanks, @bsideup ,

I look forward to that soon, and appreciate all that you do!

**Comment by bsideup** [comment]
Hi @haicoder,

Sorry, we had an issue in our CI system, but now it is fixed and the SNAPSHOT should be published. Could you please try the latest snapshot? see README for the details

**Comment by haicoder** [comment]
Hi @bsideup,

I tried that snapshot, and started my application but I also got an exception : 
```
[root@localhost test]# /usr/java/jdk1.8.0_201-amd64/bin/java -jar demo.jar
10:16:19.510 [main] DEBUG reactor.util.Loggers$LoggerFactory - Using Slf4j logging framework
reactor.shaded.javassist.CannotCompileException: [source error] no such class: reactor.BlockHoundRuntime
	at reactor.shaded.javassist.CtBehavior.insertBefore(CtBehavior.java:806)
	at reactor.shaded.javassist.CtBehavior.insertBefore(CtBehavior.java:766)
	at reactor.BlockHound$BlockingClassFileTransformer.transform(BlockHound.java:354)
	at sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:428)
	at sun.instrument.InstrumentationImpl.retransformClasses0(Native Method)
	at sun.instrument.InstrumentationImpl.retransformClasses(InstrumentationImpl.java:144)
	at reactor.BlockHound$Builder.install(BlockHound.java:265)
	at reactor.BlockHound.install(BlockHound.java:69)
	at com.example.demo.DemoApplication.main(DemoApplication.java:16)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
Caused by: compile error: no such class: reactor.BlockHoundRuntime
	at reactor.shaded.javassist.compiler.MemberResolver.searchImports(MemberResolver.java:479)
	at reactor.shaded.javassist.compiler.MemberResolver.lookupClass(MemberResolver.java:422)
	at reactor.shaded.javassist.compiler.MemberResolver.lookupClassByJvmName(MemberResolver.java:329)
	at reactor.shaded.javassist.compiler.TypeChecker.atCallExpr(TypeChecker.java:711)
	at reactor.shaded.javassist.compiler.JvstTypeChecker.atCallExpr(JvstTypeChecker.java:170)
	at reactor.shaded.javassist.compiler.ast.CallExpr.accept(CallExpr.java:49)
	at reactor.shaded.javassist.compiler.CodeGen.doTypeCheck(CodeGen.java:266)
	at reactor.shaded.javassist.compiler.CodeGen.atStmnt(CodeGen.java:360)
	at reactor.shaded.javassist.compiler.ast.Stmnt.accept(Stmnt.java:53)
	at reactor.shaded.javassist.compiler.CodeGen.atStmnt(CodeGen.java:381)
	at reactor.shaded.javassist.compiler.ast.Stmnt.accept(Stmnt.java:53)
	at reactor.shaded.javassist.compiler.Javac.compileStmnt(Javac.java:578)
	at reactor.shaded.javassist.CtBehavior.insertBefore(CtBehavior.java:786)
	... 17 more
reactor.shaded.javassist.CannotCompileException: [source error] no such class: reactor.BlockHoundRuntime
	at reactor.shaded.javassist.CtBehavior.insertBefore(CtBehavior.java:806)
	at reactor.shaded.javassist.CtBehavior.insertBefore(CtBehavior.java:766)
	at reactor.BlockHound$BlockingClassFileTransformer.transform(BlockHound.java:354)
	at sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:428)
	at sun.instrument.InstrumentationImpl.retransformClasses0(Native Method)
	at sun.instrument.InstrumentationImpl.retransformClasses(InstrumentationImpl.java:144)
	at reactor.BlockHound$Builder.install(BlockHound.java:265)
	at reactor.BlockHound.install(BlockHound.java:69)
	at com.example.demo.DemoApplication.main(DemoApplication.java:16)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
Caused by: compile error: no such class: reactor.BlockHoundRuntime
	at reactor.shaded.javassist.compiler.MemberResolver.lookupClass(MemberResolver.java:410)
	at reactor.shaded.javassist.compiler.MemberResolver.lookupClassByJvmName(MemberResolver.java:329)
	at reactor.shaded.javassist.compiler.TypeChecker.atCallExpr(TypeChecker.java:711)
	at reactor.shaded.javassist.compiler.JvstTypeChecker.atCallExpr(JvstTypeChecker.java:170)
	at reactor.shaded.javassist.compiler.ast.CallExpr.accept(CallExpr.java:49)
	at reactor.shaded.javassist.compiler.CodeGen.doTypeCheck(CodeGen.java:266)
	at reactor.shaded.javassist.compiler.CodeGen.atStmnt(CodeGen.java:360)
	at reactor.shaded.javassist.compiler.ast.Stmnt.accept(Stmnt.java:53)
	at reactor.shaded.javassist.compiler.CodeGen.atStmnt(CodeGen.java:381)
	at reactor.shaded.javassist.compiler.ast.Stmnt.accept(Stmnt.java:53)
	at reactor.shaded.javassist.compiler.Javac.compileStmnt(Javac.java:578)
	at reactor.shaded.javassist.CtBehavior.insertBefore(CtBehavior.java:786)
	... 17 more
reactor.shaded.javassist.CannotCompileException: [source error] no such class: reactor.BlockHoundRuntime
	at reactor.shaded.javassist.CtBehavior.insertBefore(CtBehavior.java:806)
	at reactor.shaded.javassist.CtBehavior.insertBefore(CtBehavior.java:766)
	at reactor.BlockHound$BlockingClassFileTransformer.transform(BlockHound.java:354)
	at sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:428)
	at sun.instrument.InstrumentationImpl.retransformClasses0(Native Method)
	at sun.instrument.InstrumentationImpl.retransformClasses(InstrumentationImpl.java:144)
	at reactor.BlockHound$Builder.install(BlockHound.java:265)
	at reactor.BlockHound.install(BlockHound.java:69)
	at com.example.demo.DemoApplication.main(DemoApplication.java:16)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
Caused by: compile error: no such class: reactor.BlockHoundRuntime
	at reactor.shaded.javassist.compiler.MemberResolver.lookupClass(MemberResolver.java:410)
	at reactor.shaded.javassist.compiler.MemberResolver.lookupClassByJvmName(MemberResolver.java:329)
	at reactor.shaded.javassist.compiler.TypeChecker.atCallExpr(TypeChecker.java:711)
	at reactor.shaded.javassist.compiler.JvstTypeChecker.atCallExpr(JvstTypeChecker.java:170)
	at reactor.shaded.javassist.compiler.ast.CallExpr.accept(CallExpr.java:49)
	at reactor.shaded.javassist.compiler.CodeGen.doTypeCheck(CodeGen.java:266)
	at reactor.shaded.javassist.compiler.CodeGen.atStmnt(CodeGen.java:360)
	at reactor.shaded.javassist.compiler.ast.Stmnt.accept(Stmnt.java:53)
	at reactor.shaded.javassist.compiler.CodeGen.atStmnt(CodeGen.java:381)
	at reactor.shaded.javassist.compiler.ast.Stmnt.accept(Stmnt.java:53)
	at reactor.shaded.javassist.compiler.Javac.compileStmnt(Javac.java:578)
	at reactor.shaded.javassist.CtBehavior.insertBefore(CtBehavior.java:786)
	... 17 more
reactor.shaded.javassist.CannotCompileException: [source error] no such class: reactor.BlockHoundRuntime
	at reactor.shaded.javassist.CtBehavior.insertBefore(CtBehavior.java:806)
	at reactor.shaded.javassist.CtBehavior.insertBefore(CtBehavior.java:766)
	at reactor.BlockHound$BlockingClassFileTransformer.transform(BlockHound.java:354)
	at sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:428)
	at sun.instrument.InstrumentationImpl.retransformClasses0(Native Method)
	at sun.instrument.InstrumentationImpl.retransformClasses(InstrumentationImpl.java:144)
	at reactor.BlockHound$Builder.install(BlockHound.java:265)
	at reactor.BlockHound.install(BlockHound.java:69)
	at com.example.demo.DemoApplication.main(DemoApplication.java:16)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
Caused by: compile error: no such class: reactor.BlockHoundRuntime
	at reactor.shaded.javassist.compiler.MemberResolver.lookupClass(MemberResolver.java:410)
	at reactor.shaded.javassist.compiler.MemberResolver.lookupClassByJvmName(MemberResolver.java:329)
	at reactor.shaded.javassist.compiler.TypeChecker.atCallExpr(TypeChecker.java:711)
	at reactor.shaded.javassist.compiler.JvstTypeChecker.atCallExpr(JvstTypeChecker.java:170)
	at reactor.shaded.javassist.compiler.ast.CallExpr.accept(CallExpr.java:49)
	at reactor.shaded.javassist.compiler.CodeGen.doTypeCheck(CodeGen.java:266)
	at reactor.shaded.javassist.compiler.CodeGen.atStmnt(CodeGen.java:360)
	at reactor.shaded.javassist.compiler.ast.Stmnt.accept(Stmnt.java:53)
	at reactor.shaded.javassist.compiler.CodeGen.atStmnt(CodeGen.java:381)
	at reactor.shaded.javassist.compiler.ast.Stmnt.accept(Stmnt.java:53)
	at reactor.shaded.javassist.compiler.Javac.compileStmnt(Javac.java:578)
	at reactor.shaded.javassist.CtBehavior.insertBefore(CtBehavior.java:786)
	... 17 more
reactor.shaded.javassist.CannotCompileException: [source error] no such class: reactor.BlockHoundRuntime
	at reactor.shaded.javassist.CtBehavior.insertBefore(CtBehavior.java:806)
	at reactor.shaded.javassist.CtBehavior.insertBefore(CtBehavior.java:766)
	at reactor.BlockHound$BlockingClassFileTransformer.transform(BlockHound.java:354)
	at sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:428)
	at sun.instrument.InstrumentationImpl.retransformClasses0(Native Method)
	at sun.instrument.InstrumentationImpl.retransformClasses(InstrumentationImpl.java:144)
	at reactor.BlockHound$Builder.install(BlockHound.java:265)
	at reactor.BlockHound.install(BlockHound.java:69)
	at com.example.demo.DemoApplication.main(DemoApplication.java:16)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
Caused by: compile error: no such class: reactor.BlockHoundRuntime
	at reactor.shaded.javassist.compiler.MemberResolver.lookupClass(MemberResolver.java:410)
	at reactor.shaded.javassist.compiler.MemberResolver.lookupClassByJvmName(MemberResolver.java:329)
	at reactor.shaded.javassist.compiler.TypeChecker.atCallExpr(TypeChecker.java:711)
	at reactor.shaded.javassist.compiler.JvstTypeChecker.atCallExpr(JvstTypeChecker.java:170)
	at reactor.shaded.javassist.compiler.ast.CallExpr.accept(CallExpr.java:49)
	at reactor.shaded.javassist.compiler.CodeGen.doTypeCheck(CodeGen.java:266)
	at reactor.shaded.javassist.compiler.CodeGen.atStmnt(CodeGen.java:360)
	at reactor.shaded.javassist.compiler.ast.Stmnt.accept(Stmnt.java:53)
	at reactor.shaded.javassist.compiler.CodeGen.atStmnt(CodeGen.java:381)
	at reactor.shaded.javassist.compiler.ast.Stmnt.accept(Stmnt.java:53)
	at reactor.shaded.javassist.compiler.Javac.compileStmnt(Javac.java:578)
	at reactor.shaded.javassist.CtBehavior.insertBefore(CtBehavior.java:786)
	... 17 more
reactor.shaded.javassist.CannotCompileException: [source error] no such class: reactor.BlockHoundRuntime
	at reactor.shaded.javassist.CtBehavior.insertBefore(CtBehavior.java:806)
	at reactor.shaded.javassist.CtBehavior.insertBefore(CtBehavior.java:766)
	at reactor.BlockHound$BlockingClassFileTransformer.transform(BlockHound.java:354)
	at sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:428)
	at sun.instrument.InstrumentationImpl.retransformClasses0(Native Method)
	at sun.instrument.InstrumentationImpl.retransformClasses(InstrumentationImpl.java:144)
	at reactor.BlockHound$Builder.install(BlockHound.java:265)
	at reactor.BlockHound.install(BlockHound.java:69)
	at com.example.demo.DemoApplication.main(DemoApplication.java:16)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
Caused by: compile error: no such class: reactor.BlockHoundRuntime
	at reactor.shaded.javassist.compiler.MemberResolver.lookupClass(MemberResolver.java:410)
	at reactor.shaded.javassist.compiler.MemberResolver.lookupClassByJvmName(MemberResolver.java:329)
	at reactor.shaded.javassist.compiler.TypeChecker.atCallExpr(TypeChecker.java:711)
	at reactor.shaded.javassist.compiler.JvstTypeChecker.atCallExpr(JvstTypeChecker.java:170)
	at reactor.shaded.javassist.compiler.ast.CallExpr.accept(CallExpr.java:49)
	at reactor.shaded.javassist.compiler.CodeGen.doTypeCheck(CodeGen.java:266)
	at reactor.shaded.javassist.compiler.CodeGen.atStmnt(CodeGen.java:360)
	at reactor.shaded.javassist.compiler.ast.Stmnt.accept(Stmnt.java:53)
	at reactor.shaded.javassist.compiler.CodeGen.atStmnt(CodeGen.java:381)
	at reactor.shaded.javassist.compiler.ast.Stmnt.accept(Stmnt.java:53)
	at reactor.shaded.javassist.compiler.Javac.compileStmnt(Javac.java:578)
	at reactor.shaded.javassist.CtBehavior.insertBefore(CtBehavior.java:786)
	... 17 more

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::        (v2.1.3.RELEASE)

2019-03-12 10:16:21.063  INFO 80893 --- [           main] com.example.demo.DemoApplication         : Starting DemoApplication v0.0.1-SNAPSHOT on localhost.localdomain with PID 80893 (/tmp/test/demo.jar started by root in /tmp/test)
2019-03-12 10:16:21.065  INFO 80893 --- [           main] com.example.demo.DemoApplication         : No active profile set, falling back to default profiles: default
2019-03-12 10:16:24.233  INFO 80893 --- [           main] o.s.b.web.embedded.netty.NettyWebServer  : Netty started on port(s): 8080
2019-03-12 10:16:24.239  INFO 80893 --- [           main] com.example.demo.DemoApplication         : Started DemoApplication in 3.78 seconds (JVM running for 5.491)
```

And my pom.xml and codes:
```
	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>2.1.3.RELEASE</version>
		<relativePath/> <!-- lookup parent from repository -->
	</parent>
	<repositories>
		<repository>
			<id>spring-snapshots</id>
			<name>Spring Snapshot Repository</name>
			<url>https://repo.spring.io/snapshot</url>
		</repository>
	</repositories>
	<dependencies>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-webflux</artifactId>
		</dependency>
		<dependency>
			<groupId>io.projectreactor.tools</groupId>
			<artifactId>blockhound</artifactId>
			<version>1.0.0.BUILD-SNAPSHOT</version>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-test</artifactId>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>io.projectreactor</groupId>
			<artifactId>reactor-test</artifactId>
			<scope>test</scope>
		</dependency>
	</dependencies>
```
```
public static void main(String[] args) {
	BlockHound.install();
        Mono.delay(Duration.ofMillis(1))
            .doOnNext(it -> {
                try {
                    Thread.sleep(10);
                }
                catch (InterruptedException e) {
                    throw new RuntimeException(e);
                }
            })
            .block(); // should throw an exception about Thread.sleep
        System.setProperty("reactor.netty.http.server.accessLogEnabled", "true");
	SpringApplication.run(DemoApplication.class, args);
}
```

Am I using it correctly? And whether this error matters.

**Comment by bsideup** [comment]
Hi @haicoder,

Thanks a lot for trying again and providing the sample! I can confirm that it fails on my machine too when I use Java 8.
I will investigate the failure that report back ASAP.

**Comment by bsideup** [comment]
@haicoder,

Thanks a lot for your report, it helped to pinpoint the problem and now it should be fixed in the latest SNAPSHOT 👍

**Comment by haicoder** [comment]
Hi @bsideup ,

Nice!  Works fine on my machine, and it detects some blocking calls in my application.

Thanks a lot for your help : )

**Comment by bsideup** [comment]
@haicoder I'm so happy that the issue is solved and it works for you 👍
I'll close this issue as it is fixed now, but feel free to report more if you discover.

**Comment by bsideup** [comment]
Released in 1.0.0.M2 🎉
